### PR TITLE
feat(dropdown): Add samples with icon/subtitle/context/menu

### PR DESCRIPTION
**Pull request by yasincaliskan**
### Description
- Dropdown subtitle CSS class has `font-size` properties smaller than the title class.
- Dropdown samples have separated as `States`, `No Options`, `Pending Request`, `Disabled`.

### New Samples
- Dropdown Menu with icon
- Dropdown options with subtitle
- Dropdown with customized options

### Related Issue
- https://github.com/Hipo/react-ui-toolkit/issues/50

### CODE REVIEW

File: stories/1-Dropdown.stories.tsx [hunk 1]
```diff
@@ -55,40 +98,146 @@ storiesOf("Dropdown", module).add("Dropdown States", () => {
...
             />
           </FormField>
-
-          <FormField label={"Your Language"}>
+        </div>
+      )}
+    </StateProvider>
+  ))
+  .add("Dropdown With Customized Options", () => (
+    <StateProvider initialState={initialStateWithContext}>
```
Comment (edizcelik) [inline stories/1-Dropdown.stories.tsx:169]: This doesn't show how one would use "Customized Options" but shows the use of `context`

File: stories/1-Dropdown.stories.tsx [hunk 2]
```diff
@@ -1,35 +1,78 @@
...
-      },
-      {
-        id: "spanish",
-        title: "Spanish"
-      },
-      {
-        id: "french",
-        title: "French - Disabled",
-        isDisabled: true
+const initialState = {
```
Comment (edizcelik) [inline stories/1-Dropdown.stories.tsx:9]: Let's define the types for these variables. Also lets try to move them to a separate module, e.g. `dropdownStoryConstants`

File: stories/1-Dropdown.stories.tsx [hunk 3]
```diff
@@ -55,40 +98,146 @@ storiesOf("Dropdown", module).add("Dropdown States", () => {
...
+                          border-radius: 50%;
+                          display: inline-block;
+                        }
+                        
+                        .dropdown-header {
+                          color: black;
+                          margin-left: 16px;
+                        }
+                        `}
+                    </style>
```
Comment (edizcelik) [inline stories/1-Dropdown.stories.tsx:137]: There should be a better way of doing this

**Review by edizcelik — CHANGES_REQUESTED** [review]

**Review by bilgeyucel — APPROVED** [review]
I really liked the examples you provided. Seems neat, thanks 💯

**Comment by yasincaliskan** [comment]
Thank you so much! 🎉

**Review by jamcry — APPROVED** [review]

**Review by Anlerkan — APPROVED** [review]

**Review by mucahit — APPROVED** [review]

**Review by edizcelik — APPROVED** [review]

**Pull request merged**